src/Aquifer.Jobs/Subscribers/NotificationSubscriber.cs

```diff
@@ -0,0 +1,174 @@
+                    ["projectId"] = project.Id,
+                    ["projectName"] = project.Name,
+                },
+                Tos: [NotificationsHelper.NotificationToEmailAddress],
+                Bccs: [NotificationsHelper.GetEmailAddress(companyLeadUser)]);
+
+            await _emailService.SendEmailAsync(templatedEmail, ct);
+
+            _logger.LogInformation(
+                "Project started notification sent for Project ID {ProjectId} to User ID {UserId}.",
+                project.Id,
+                companyLeadUser.Id);
+        }
+        else
+        {
+            _logger.LogWarning(
+                "Unable to send project started notification for Project ID {ProjectId} because there is no assigned Company or Company Lead.",
+                project.Id);
+        }
+    }
+
+    [Function(nameof(SendResourceCommentCreatedNotification))]
+    public async Task SendResourceCommentCreatedNotification(
+        [QueueTrigger(Queues.SendResourceCommentCreatedNotification)] QueueMessage queueMessage,
+        CancellationToken ct)
+    {
+        var message = queueMessage.Deserialize<SendResourceCommentCreatedNotificationMessage, NotificationSubscriber>(_logger);
+
+        var dbConnection = _dbContext.Database.GetDbConnection();
+        var resourceCommentData = await GetResourceCommentDataAsync(dbConnection, message.CommentId, ct);
+
+        // Get users who are assigned or have previously been assigned to this resource
+        // and who are in the same company as the user who made the comment,
+        // excluding the user who made the comment (who should not get a notification).
+        var previouslyAssignedUsers = await _dbContext.ResourceContentVersionAssignedUserHistory
+            .Where(rcvauh =>
+                rcvauh.ResourceContentVersion.ResourceContentId == resourceCommentData.ResourceContentId &&
+                rcvauh.AssignedUserId != null &&
+                rcvauh.AssignedUserId.Value != resourceCommentData.UserId)
+            .Select(rcvauh => rcvauh.AssignedUserId)
+            .Distinct()
+            .Join(
+                _dbContext.Users
+                    .Where(u => u.Id == resourceCommentData.UserId && u.AquiferNotificationsEnabled)
+                    .SelectMany(u => u.Company.Users),
+                userId => userId,
+                u => u.Id,
+                (_, u) => u)
+            .ToListAsync(ct);
+
+        if (previouslyAssignedUsers.Count == 0)
+        {
+            _logger.LogInformation(
+                "Comment ID {CommentId} was added to Resource Content ID {ResourceContentId} by User ID {CommenterUserId} but no email was sent because only the commenter has previous resource assignment.",
+                resourceCommentData.CommentId,
+                resourceCommentData.ResourceContentId,
+                resourceCommentData.UserId);
+            return;
+        }
+
+        var commenterUser = await _dbContext.Users
```

Comment:
Would it be more efficient to join into Users from  `GetResourceCommentDataAsync` to avoid the extra database trip here?

Reply:
I'll admit that this was a product of my development process.  At one point I was fetching this user when also fetching the resource assignment users (which was the most efficient) but had to stop doing that as things were refactored.

The `GetResourceCommentDataAsync` method uses Dapper (I tried writing it in EF and got stumped even though it seems like it should be easy). I didn't want to make some kind of Dapper/EF interop mapping nor did I want to use the EF `UserEntity` as a destination object for the Dapper query because my assumption when seeing an EF entity is that it was pulled via EF and might have tracking, etc.  I also wanted a `UserEntity` in order to have common code that knows how to generate a full name.

So yes, this could be more efficiently done.  I will update to also fetch the commenter's name in the Dapper call which is really all I want for the email.